pdp/ast/yast/obligations.go

```diff
@@ -23,9 +23,20 @@ func (ctx context) unmarshalObligationItem(v interface{}) (pdp.AttributeAssignme
 		return pdp.AttributeAssignmentExpression{}, newUnknownAttributeError(ID)
 	}
 
-	e, err := ctx.unmarshalValueByType(a.GetType(), v)
-	if err != nil {
-		return pdp.AttributeAssignmentExpression{}, bindError(err, ID)
+	var e pdp.Expression
+	m, err = ctx.validateMap(v, "obligation assignment")
+	if err == nil {
+		// if it's a map, assume it's an expression
```

Comment:
Let's stick to no backward compatibility. It allows in addition to other simplifications to bypass some corner cases related to the assumption.

Reply:
Sounds good - I'll take those code out.

BTW, I just remembered - one reason I tried to make it backward compatible is because the XACML specification stipulates using immediate value. I'm good if you don't feel that's critical.

Thanks!!